### PR TITLE
Removes deprecated class Piwik\Translate

### PR DESCRIPTION
**Pull request by sgiehl**
refs #8567

### CODE REVIEW

**Comment by tsteur** [comment]
BTW: I'm wondering in that case it it's worth removing the class? Not sure there is much of a down side to keep it and at the same time it might force heaps of plugins to change the code maybe? Might be the same for some other classes /methods. Wondering if sometimes we could just keep the old classes for simplicity of BC. In this case it wouldn't hurt us much maybe to keep that old class? 

Of course we'd still ideally change some usages to be a good example and make use of it correctly ideally. 

In general since we manage > 50 plugins it might be easier to sometimes just keep the old classes if they are as simple as the translate class

File: CHANGELOG.md [hunk 1]
```diff
@@ -30,6 +30,7 @@ The Product Changelog at **[matomo.org/changelog](https://matomo.org/changelog)*
 * The deprecated Platform API class `\Piwik\Registry` has been removed. Use `\Piwik\Container\StaticContainer` instead
 * The deprecated Platform API class `\Piwik\TaskScheduler` has been removed. Use `\Piwik\Scheduler\Scheduler` instead
 * The deprecated Platform API class `\Piwik\DeviceDetectorFactory` has been removed. Use `\Piwik\DeviceDetector\DeviceDetectorFactory` instead
+* The deprecated Platform API class `\Piwik\Translate` has been remoced. Use `\Piwik\Translation\Translator` instead.
```
Comment (tsteur) [inline CHANGELOG.md]: btw typo in remoced

**Comment by sgiehl** [comment]
We already deprecated that class in Piwik 2.11 and kept it in 3.x for BC. At some point we need to break BC and remove such stuff otherwise we will mess up the code with more and more such stuff :man_shrugging:

File: CHANGELOG.md (hunk 1 above)
Comment (sgiehl) [reply CHANGELOG.md]:
```suggestion
* The deprecated Platform API class `\Piwik\Translate` has been removed. Use `\Piwik\Translation\Translator` instead.
```

**Comment by tsteur** [comment]
>  At some point we need to break BC and remove such stuff otherwise we will mess up the code with more and more such stuff 🤷‍♂

Not sure how one file would mess up the code there? Of course it always depends but in this case not seeing it so much. There doesn't seem to be much of a harm as it's only forwarding methods anyway. If it helps keeping plugins easily compatible and less pain for us and developers be great. Personally, I love eg how Android and iOS etc keep BC basically almost forever (not saying we should always do this but when it's easy like there might be worth it)

**Comment by sgiehl** [comment]
OK. What about keeping that class but triggering a deprecation notice
whenever a method is called and remove it in Matomo 5 then?

>

**Comment by tsteur** [comment]
Not sure. I reckon in that case there might not even be a need to remove it in Matomo 5 either. I'm thinking if we can keep BC whenever it's easy for us we probably should. Could even remove the deprecated flag there. In development mode we could trigger a notice though maybe and recommend using a different way. 👍

**Comment by sgiehl** [comment]
@tsteur readded the file, triggering deprecation notices in development mode

**Comment by tsteur** [comment]
👍

File: core/Translate.php [hunk 2]
```diff
@@ -123,7 +133,15 @@ private static function getTranslator()
...
     {
+        self::triggerDeprecationNotice();
         self::loadCoreTranslation();
         Manager::getInstance()->loadPluginTranslations();
     }
+
+    protected static function triggerDeprecationNotice()
+    {
+        if (Development::isEnabled()) {
+            trigger_error('Using \\Piwik\\Translate is deprecated. Use \\Piwik\\Translation\\Translator instead.', E_USER_DEPRECATED);
```
Comment (tsteur) [inline core/Translate.php]: btw any reason we're not using Log::warn or so?

Comment (sgiehl) [reply core/Translate.php]: No. Guess a warning should be fine as well, as it's only in development mode

**Comment by sgiehl** [comment]
@tsteur let me know if you think it's ready to merge. Would do that in all linked repos then and update the submodules before merge.

**Comment by tsteur** [comment]
@sgiehl I haven't really looked (only very quickly) but reckon it's  fine 👍 Unfortunately won't be able to do any reviews in the next 7 days :(

**Pull request merged**